File: ShadowFrameworkLite20/src/shadow/image/SFFilteredRenderedTexture.java
package shadow.image;

import shadow.pipeline.SFPipeline;
import shadow.pipeline.SFPipelineGraphics.Module;
import shadow.pipeline.SFProgram;
import shadow.renderer.SFProgramModuleStructures;
import shadow.renderer.SFRenderer;

public class SFFilteredRenderedTexture implements SFRenderedTexturesSet{

	private boolean initialized=false;
	private SFPipelineTexture[] textures;
	private SFRenderedTexture renderedTexture=new SFRenderedTexture();
	private SFProgramModuleStructures lightComponent=new SFProgramModuleStructures();
	private SFProgramModuleStructures material=new SFProgramModuleStructures();
	

	public SFProgramModuleStructures getMaterialComponent() {
		return material;
	}

	public void setMaterialComponent(SFProgramModuleStructures material) {
		this.material = material;
	}

	@Override
	public SFPipelineTexture getTexture(int index) {
		return textures[index];
	}
	
	@Override
	public int getTextureSize() {
		return textures.length;
	}
	
	@Override
	public void init() {
		
		if(!initialized){
			for (int i = 0; i < textures.length; i++) {
				SFPipelineTexture data=textures[i];
				textures[i]=SFPipeline.getSfTexturePipeline().getRenderedTextureFactory().generateTextureBuffer(data.getWidth(), data.getHeight(), 
						data.getFormat(),  data.getFilters(),
						data.getWrapS(), data.getWrapT());
				renderedTexture.addColorData(textures[i]);
			}
			update();
			
		}
		initialized=true;
	}
	
	
	
	@Override
	public void update() {

		SFPipeline.getSfTexturePipeline().beginNewRenderedTexture(renderedTexture);

			SFProgram program=SFPipeline.getStaticImageProgram(getMaterialComponent().getProgram(),lightComponent.getProgram());
			SFPipeline.getSfProgramBuilder().loadProgram(program);
		
			SFRenderer.setupMaterialData(Module.MATERIAL,material);
			SFRenderer.setupMaterialData(Module.LIGHT,lightComponent);
			
			SFPipeline.getSfPipelineGraphics().drawBaseQuad();
			
		SFPipeline.getSfTexturePipeline().endRenderedTexture(renderedTexture);
		
		SFPipeline.getSfTexturePipeline().destroyRenderedTexture(renderedTexture);
	}
	
	@Override
	public void destroy() {
		if(initialized){
			for (int i = 0; i < textures.length; i++) {
				SFPipeline.getSfTexturePipeline().getRenderedTextureFactory().destroyBuffer(textures[i]);
			}
		}
	}
	
	public SFRenderedTexture getRenderedTexture() {
		return renderedTexture;
	}

	public void setTextures(SFPipelineTexture[] textures) {
		this.textures = textures;
	}

	public SFProgramModuleStructures getLightComponent() {
		return lightComponent;
	}
	
	public void setLightComponent(SFProgramModuleStructures lightComponent) {
		this.lightComponent = lightComponent;
	}

	
	
}

File: trunk/ShadowFrameworkJS/src/shadow/image/SFFilteredRenderedTexture.js


function SFFilteredRenderedTexture(){
	this.initialized=false;
	this.textures=null;
	this.renderedTexture=new SFRenderedTexture();
	this.lightComponent=new SFProgramModuleStructures();
	this.material=new SFProgramModuleStructures();
}


SFFilteredRenderedTexture.prototype["getMaterialComponent"]=function(){
	return this.material;
};


SFFilteredRenderedTexture.prototype["getTexture"]=function(index){
	return this.textures[index];
};


SFFilteredRenderedTexture.prototype["getTextureSize"]=function(){
	return this.textures.length;
};

SFFilteredRenderedTexture.prototype["setMaterialComponent"]=function(material){
	this.material = material;
};

SFFilteredRenderedTexture.prototype["init"]=function(){
	
		if(!this.initialized){
			for (var i = 0; i < this.textures.length; i++) {
				var data=this.textures[i];
				this.textures[i]=SFPipeline_getSfTexturePipeline().getRenderedTextureFactory().
						generateTextureBuffer(data.getWidth(), data.getHeight(), 
						data.getFormat(),  data.getFilters(),
						data.getWrapS(), data.getWrapT());
				this.renderedTexture.addColorData(this.textures[i]);
			}
			this.update();
			
		}
		initialized=true;
};
	
	
SFFilteredRenderedTexture.prototype["update"]=function(){
	
		
		SFPipeline_getSfTexturePipeline().beginNewRenderedTexture(this.renderedTexture);

			var program=SFPipeline_getStaticImageProgram(this.getMaterialComponent().getProgram(),
				this.lightComponent.getProgram());
			SFPipeline_getSfProgramBuilder().loadProgram(program);
		
			SFRenderer_setupMaterialData(SFPipelineGraphics_Module_MATERIAL,this.material);
			SFRenderer_setupMaterialData(SFPipelineGraphics_Module_LIGHT,this.lightComponent);
			
			SFPipeline_getSfPipelineGraphics().drawBaseQuad();
			
		SFPipeline_getSfTexturePipeline().endRenderedTexture(this.renderedTexture);
		
		SFPipeline_getSfTexturePipeline().destroyRenderedTexture(this.renderedTexture);
};
	
	
SFFilteredRenderedTexture.prototype["destroy"]=function(){
	if(this.initialized){
		for (var i = 0; i < this.textures.length; i++) {
			SFPipeline_getSfTexturePipeline().getRenderedTextureFactory().destroyBuffer(this.textures[i]);
		}
	}
};
	
SFFilteredRenderedTexture.prototype["getRenderedTexture"]=function(){
		return this.renderedTexture;
};
		
SFFilteredRenderedTexture.prototype["getLightComponent"]=function(){
		return this.lightComponent;
};
	
SFFilteredRenderedTexture.prototype["setTextures"]=function(textures){
		this.textures = textures;
};
	
SFFilteredRenderedTexture.prototype["setLightComponent"]=function(lightComponent){
		this.lightComponent = lightComponent;
};
	

	